project_1/src/simulator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "simulator.hpp"
+
+#include <chrono>
+
+NetworkSimulator::NetworkSimulator(const std::vector<std::uint32_t>& distances_to_receiver) : servers_number(distances_to_receiver.size()) {
+    for (std::uint32_t distance: distances_to_receiver) {
+        AddServer(distance);
+    }
+    std::uint32_t server_receiver_id = 0;
+    server_receiver = ServerBase(server_receiver_id);
+}
+
+void NetworkSimulator::AddServer(std::uint32_t distance_to_receiver) {
+    if (servers.size() < servers_number) {
+        servers.emplace_back(GenerateNewID(servers), distance_to_receiver);
+    }
+}
+
+void NetworkSimulator::StartSimulation() {
+    std::uint32_t test_packages_number = 5;
+
+    // Each server send 5 packages
+    for (ServerSender& server: servers) {
+        // Create event
+        events.emplace_back(GenerateNewID(events), server, SEND_DATA, test_packages_number);
+
+        // Create packages, add them to priority_queue
+        for (std::uint32_t i = 0; i < test_packages_number; ++i) {
+            auto current_time_μs = GetCurrentTimeInMicroseconds();
+            packages.emplace(server.GetID(), server_receiver.GetID(), current_time_μs, i);
+        }
+        std::cout << events.back();
+    }
+
+    auto current_time_μs = GetCurrentTimeInMicroseconds();
+    std::uint32_t sliding_window_duration_μs = 1'000'000;
```

Comment:
Just for interest, how did you manage to type `μs`? Still, I suggest not to use special characters like `μ` and replace it by standard Latin letters to avoid possible font issues and better portability. It is common to use `us` notation for `μs` (as `u` looks similar to `μ`).

Reply:
I agree, it was a bad idea. I just copied `μ` symbol from some website, will use `u` instead.